bindings/java/src/test/java/org/apache/opendal/test/NativeLayerTest.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.opendal.test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.Cleanup;
+import org.apache.opendal.NativeLayer;
+import org.apache.opendal.Operator;
+import org.apache.opendal.layer.RetryNativeLayer;
+import org.junit.jupiter.api.Test;
+
+public class NativeLayerTest {
+    @Test
+    void testOperatorWithRetryLayer() {
+        final Map<String, String> conf = new HashMap<>();
+        conf.put("root", "/opendal/");
+        final NativeLayer retryLayer = RetryNativeLayer.builder().build();
+        @Cleanup final Operator op = Operator.of("memory", conf);
```

Comment:
Can we have `Operator.of("memory", conf).layer(retryLayer)`?

Reply:
This is the issue I talked with you about the ownership. Since we use clone in the rust side:

```rust
Box::into_raw(Box::new(op.clone().layer(retry))) as jlong
```

If we writes `Operator.of("memory", conf).layer(retryLayer)`, the temp var `Operator.of("memory", conf)` cannot properly release its underneath resource.

Reply:
Let me have a try on this.

Reply:
I attempted but failed to improve the code. You're right in your assessment!